api/storage_service.cc

```diff
@@ -1409,7 +1409,7 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
                 return do_for_each(column_families, [=, &db](sstring cfname) {
                     auto& cm = db.get_compaction_manager();
                     auto& cf = db.find_column_family(keyspace, cfname);
-                    return cm.perform_sstable_scrub(&cf, opts);
+                    return cm.perform_sstable_scrub(cf.as_table_state(), opts);
```

Comment:
I think we should add a table method to do that, rather than open coding `as_table_state` here and in other layers above table.

When the table will have multiple compaction group it will need to scrub all of them anyhow (and synchronize with compaction group changes).

Reply:
I agree, but I think it's best deferred to a later series and let this one focus on separating compaction/ and replica/.

Reply:
Yes, when implementing this, it became clear that for multiple groups, we'll need replica::table being a bridge to all of its groups, so if api/storage_service wants to perform an operation which spans all groups, then replica::table will call manager on behalf of all groups, and only then return to the caller. And yes, the idea is that this series is the first step towards group. once this series is in, then we'll need to do work on table for supporting multiple groups.